tests/postgres_tests/models.py

```diff
@@ -177,3 +179,24 @@ class NowTestModel(models.Model):
 
 class UUIDTestModel(models.Model):
     uuid = models.UUIDField(default=None, null=True)
+
+
+class Room(PostgreSQLModel):
+    number = models.IntegerField(null=False, unique=True)
+    no_people = models.IntegerField(null=False)
+
+
+class Reservation(models.Model):
+    room = models.ForeignKey('Room', on_delete=models.CASCADE)
+    timespan = DateRangeField(null=False)
+    cancelled = models.BooleanField(default=False)
+
+    class Meta:
+        constraints = [
+            ExclusionConstraint(
+                name='exclude_overlapping_reservations',
+                expressions=[(F('timespan'), '&&'), (F('room'), '&')],
```

Comment:
Do we expect users to use strings like `&&` and `&` directly anywhere else? It seems like we're exposing an implementation detail we'd usually hide.

Reply:
In `django.contrib.postgres`, the lookups hide them inside the `operator` attribute or in the `as_sql`-method. The list of operators is not particularly long: https://www.postgresql.org/docs/current/static/functions-range.html I'll see what to do what about it; something like `range_operator.overlaps` would definitely read better, and maybe also reduce an error source. At least in the above, `room` should be excluded with an equality-operator.